### PR TITLE
Add String Terms: C++

### PR DESCRIPTION
**Pull request by KTom101**
Three new string method entries for C++. Keyword 6.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by Dusch4593** [comment]
Hey @KTom101 for your next batch of changes, could you move all the `*.md` files into their own folders of the same name? For example:

~`terms/find.md`~ -> `terms/find/find.md`

**Comment by KTom101** [comment]
@Dusch4593 I've moved the files! Let me know if there is anything else I've missed!

**Comment by KTom101** [comment]
@SSwiniarski I've made the changes except for the ones that said missing output. All three of those examples are code bytes so the output should render in the window. Were you thinking those examples should be converted from codebyte examples to regular examples?

**Comment by SSwiniarski** [comment]
Actually, I missed they were codebyte examples. They're probably fine as is.

**Review by SSwiniarski — APPROVED** [review]
Good to go for review two.

File: content/cpp/concepts/strings/terms/find/find.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+**Note:** `string::npos` will often be shown as the highest possible value of type `size_t`.
+
+## Syntax
+
+```pseudo
+stringName.find(sequence, position, count)
+```
+
+- sequence: the char or string to search for
```
Comment (Dusch4593) [inline content/cpp/concepts/strings/terms/find/find.md:25]:
```suggestion

The following arguments are a applied to the `.find()` method:

- A `sequence` representing a char or string to be search for.
```

File: content/cpp/concepts/strings/terms/size/size.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Syntax
+
+```pseudo
+stringName.size()
+```
+
+## Codebyte Example
+
+In the example below, `.size()` is called on the string variable, `bird`:
```
Comment (Dusch4593) [inline content/cpp/concepts/strings/terms/size/size.md:25]:
```suggestion
In the example below, `.size()` is called on the `bird` string:
```

File: content/cpp/concepts/strings/terms/length/length.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+## Syntax
+
+```pseudo
+stringName.length()
+```
+
+## Codebyte Example
+
+In the example below, `.length()` is called on the string variable, `snack`:
```
Comment (Dusch4593) [inline content/cpp/concepts/strings/terms/length/length.md:25]:
```suggestion
In the example below, `.length()` is called on the `snack` string:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@KTom101  took a look a finished my review! I left a few comments for you to check out 😄 

EDIT: Sorry for the goof-up, Kyra and Steve! 😢

File: content/cpp/concepts/strings/terms/length/length.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  - 'Game Development'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The `.length()` method finds the length of the string it is called on, in terms of bytes. This is the same functionality as the [`.size()`](https://www.codecademy.com/resources/docs/python/strings/size) method.
```
[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/length/length.md:15]: @Dusch4593 What do you suggest? In the documentation, they have the exact same definition. It was made because in speaking terminology some people say the size of something while others say the length of something. So the developers thought it made sense to just have both.

[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/cpp/concepts/strings/terms/length/length.md:15]: Maybe something like "While the [`.size()`](..) method is similar and returns the same value, `.length()` is commonly used with characters and strings of words and sentences." (but less wordy than what I just typed lol)

I got the idea from reading [this excerpt](https://www.geeksforgeeks.org/5-different-methods-find-length-string-c/#:~:text=The%20size()%20function%20is,to%20make%20things%20more%20readable.) from a GeeksforGeeks article.

What do you think?

File: content/cpp/concepts/strings/terms/find/find.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+}
+```
+
+Output:
+
+```shell
+18446744073709551615
+```
+
+The output `18446744073709551615` represents `string::npos`. This means the sequence, `izz`, could not be found with the parameters given. `izz` is at index 1 which comes before the specified start position of `2`. Therefore, it is outside of the search range and is not found.
```
[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/find/find.md:67]: @Dusch4593 I'm going to rework this remove the word  "our".

[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/cpp/concepts/strings/terms/find/find.md:64]: Changed!

[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/find/find.md:64]: @Dusch4593 I can change the second "our" to "the" too?

[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/cpp/concepts/strings/terms/find/find.md:64]: Yup!

File: content/cpp/concepts/strings/terms/length/length.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/length/length.md:15]: @Dusch4593 We could add it. I'm thinking of a way to say it without being too wordy.

[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/length/length.md:15]: How about this:
```suggestion
The .length() method returns the length of the given string in terms of bytes. It functions exactly like the [.size()](https://www.codecademy.com/resources/docs/python/strings/size) method but follows a more intuitive terminology, i.e., "the length of a string" rather than "the size of a string." However, both methods yield the same results.
```

File: content/cpp/concepts/strings/terms/size/size.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  - 'Game Development'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The `.size()` method finds the size of the string it is called on, in terms of bytes. The [`.length()`](https://www.codecademy.com/resources/docs/python/strings/length) method functions in the same way and can be used in place of `.size()`.
```
[the remark this answers is not in the document]
Comment (KTom101) [reply content/cpp/concepts/strings/terms/size/size.md:15]: How about:
```suggestion
The `.size()` method returns the size of a given string in terms of bytes. It functions just like the [`.length()`](https://www.codecademy.com/resources/docs/python/strings/length) method, but it's there to follow the consistency of other standard library containers (e.g., maps and [vectors](https://www.codecademy.com/resources/docs/cpp/vectors)).
```

[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/cpp/concepts/strings/terms/size/size.md:15]: Perfect!

File: content/cpp/concepts/strings/terms/find/find.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+}
+```
+
+The output for the snippet above would look like this:
+
+```shell
+18446744073709551615
+```
+
+The output `18446744073709551615` represents the `string::npos` constant to indicate that a match could not be found with the given parameters. The sequence, `izz`, begins at index 1 while the call to the `.find()` method begins at index 2. Therefore, it is outside of the search range and a matching index was not found.
```
Comment (Dusch4593) [inline content/cpp/concepts/strings/terms/find/find.md:64]:
```suggestion
The output `18446744073709551615` represents the `string::npos` constant that indicates that a match was not found with the given parameters. The sequence, `izz`, begins at index 1 while the call to the `.find()` method begins at index 2. Therefore, it is outside of the search range and a matching index was not found.
```

**Review by Dusch4593 — APPROVED** [review]
@KTom101 This looks great! I just had one last small comment/suggestion but this LGTM otherwise 👍

**Comment by KTom101** [comment]
@Dusch4593 @SSwiniarski I think I've touched all the changes. Let me know if there are any others needed. Thanks!

**Comment by SSwiniarski** [comment]
Looks fine to me.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@KTom101 Live links:

https://www.codecademy.com/resources/docs/cpp/strings/find
https://www.codecademy.com/resources/docs/cpp/strings/length
https://www.codecademy.com/resources/docs/cpp/strings/size

ty @Dusch4593 @SSwiniarski @christian-dinh 🚀
